src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalUtility.cs

```diff
@@ -79,12 +81,69 @@ public static string GetRandomEntry(string baseDir)
             return name;
         }
 
-        public static string CreateRandomDir(string baseDir)
+        public static string CreateRandomFileName(string baseDir) =>
+            RetryIO(() =>
+            {
+                string fileName = GetRandomEntry(baseDir);
+                using (File.Create(Path.Combine(baseDir, fileName)))
+                {
+                    // create new zero length file.
+                }
+                return fileName;
+            });
+
+        public static FileStream CreateRandomFileStream(string baseDir) =>
+            RetryIO(() =>
+            {
+                string fileName = GetRandomEntry(baseDir);
+                return File.Create(Path.Combine(baseDir, GetRandomEntry(baseDir)));
```

Comment:
I don't want to create file twice.

---
In reply to: [80408065](https://github.com/dotnet/docfx/pull/733#discussion_r80408065) [](ancestors = 80408065)